docs/docsite/rst/dev_guide/developing_modules.rst

```diff
@@ -1,110 +1,63 @@
 
-* Modules should not require that a user know all the underlying options of an api/tool to be used. For instance, if the legal values for a required module parameter cannot be documented, that's a sign that the module would be rejected.
+* Modules should not require that a user know all the underlying options of an API/tool to be used. For instance, if the legal values for a required module parameter cannot be documented, that's a sign that the module would be rejected.
 
 * Modules should typically encompass much of the logic for interacting with a resource. A lightweight wrapper around an API that does not contain much logic would likely cause users to offload too much logic into a playbook, and for this reason the module would be rejected. Instead try creating multiple modules for interacting with smaller individual pieces of the API.
 
+If your use case isn't covered by an existing module, an open PR, an action plugin, or a role, and you don't need to create multiple modules, then you're ready to start developing a new module. Choose from the topics below for next steps:
 
-.. _developing_modules_all:
-
-How To Develop A Module
-```````````````````````
-
-The following topics will discuss how to develop and work with modules:
-
-:doc:`developing_program_flow_modules`
-    A description of Ansible's module architecture.
-:doc:`developing_modules_general`
-    A general overview of how to develop, debug, and test modules.
-:doc:`developing_modules_general_windows`
-    A general overview of how to develop, debug and test Windows modules.
-:doc:`developing_modules_documenting`
-    How to include in-line documentation in your module.
-:doc:`developing_modules_best_practices`
-    Best practices, recommendations, and things to avoid.
-:doc:`developing_modules_checklist`
-     Checklist for contributing your module to Ansible.
-:doc:`testing`
-    Developing unit and integration tests.
-:ref:`developing_python_3`
-    Adding Python 3 support to modules (all new modules must be Python-2.6 and Python-3.5 compatible).
-:doc:`developing_modules_in_groups`
-    A guide for partners wanting to submit multiple modules.
-
+* I want to :ref:`get started on a new module <developing_modules_general>`.
+* I want to review :ref:`best practices for developing modules <developing_modules_best_practices>`.
+* I want to :ref:`write a Windows module <developing_modules_general_windows>`.
+* I want :ref:`an overview of Ansible's architecture <developing_program_flow_modules>`.
+* I want to :ref:`document my module <developing_modules_documenting>`.
+* I want to :ref:`contribute my module back to Ansible Core <developing_modules_checklist>`.
+* I want to :ref:`add unit and integration tests to my module <developing_testing>`.
+* I want to :ref:`add Python 3 support to my module <developing_python_3>`.
+* I want to :ref:`write multiple modules <developing_modules_in_groups>`.
```

Comment:
I love the phrasing on this! A very good way to direct folks to the thing they're looking for.